src/ProductConstructionService/ProductConstructionService.Api/Queue/JobsProcessorScopeManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace ProductConstructionService.Api.Queue;
+
+public enum JobsProcessorState
+{
+    // The JobsProcessor will keep taking and processing new jobs
+    Working,
+    // The JobsProcessor isn't doing anything
+    Stopped,
+    // The JobsProcessor will finish it's current job and stop
+    Stopping
+}
+
+public class JobsProcessorScopeManager
+{
+    public JobsProcessorScopeManager()
+    {
+        State = JobsProcessorState.Stopped;
+        _autoResetEvent = new(false);
+    }
```

Comment:
Maybe remove this ctor? You can initialize fields directly?

Reply:
makes sense, removed